File: WinForms/NET Framework/DemoProjectVB/Canvas/Canvas_180Rotation.vb

Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing

Imports System.Windows.Forms




Public Class Canvas_180Rotation
    Inherits Samples.BaseForm
    Private WithEvents hScrollBar2 As System.Windows.Forms.HScrollBar
    Private label1 As System.Windows.Forms.Label
    Private label2 As System.Windows.Forms.Label
    Private bar1 As Steema.TeeChart.Styles.Bar
    Private components As System.ComponentModel.IContainer = Nothing
    
    
    Public Sub New()
        ' This call is required by the Windows Form Designer.
        InitializeComponent()
        
        bar1.FillSampleValues(4)
    End Sub 'New
    
    '/ <summary>
    '/ Clean up any resources being used.
    '/ </summary>
    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub 'Dispose

#Region "Designer generated code"

    '/ <summary>
    '/ Required method for Designer support - do not modify
    '/ the contents of this method with the code editor.
    '/ </summary>
    Private Sub InitializeComponent()
        Dim resources As System.ComponentModel.ComponentResourceManager = New System.ComponentModel.ComponentResourceManager(GetType(Canvas_180Rotation))
        Me.hScrollBar2 = New System.Windows.Forms.HScrollBar
        Me.label1 = New System.Windows.Forms.Label
        Me.label2 = New System.Windows.Forms.Label
        Me.bar1 = New Steema.TeeChart.Styles.Bar
        Me.panel1.SuspendLayout()
        Me.chartContainer.SuspendLayout()
        Me.SuspendLayout()
        '
        'textBox1
        '
        Me.textBox1.Size = New System.Drawing.Size(432, 72)
        Me.textBox1.TabIndex = 0
        Me.textBox1.Text = "Chart can be rotated 180 degrees along the horizontal (x) axis." & Global.Microsoft.VisualBasic.ChrW(13) & Global.Microsoft.VisualBasic.ChrW(10) & "In previous vers" & _
            "ions it allowed 90 degree rotation only." & Global.Microsoft.VisualBasic.ChrW(13) & Global.Microsoft.VisualBasic.ChrW(10) & "Using OpenGL charts can be fully rotat" & _
            "ed 360 degree."
        '
        'panel1
        '
        Me.panel1.Controls.Add(Me.label2)
        Me.panel1.Controls.Add(Me.label1)
        Me.panel1.Controls.Add(Me.hScrollBar2)
        Me.panel1.Location = New System.Drawing.Point(0, 72)
        Me.panel1.Size = New System.Drawing.Size(432, 33)
        Me.panel1.TabIndex = 1
        '
        'tChart1
        '
        '
        '
        '
        Me.tChart1.Aspect.Chart3DPercent = 25
        Me.tChart1.Aspect.Elevation = 356
        Me.tChart1.Aspect.ElevationFloat = 356
        Me.tChart1.Aspect.Orthogonal = False
        Me.tChart1.Aspect.Perspective = 75
        Me.tChart1.Aspect.Rotation = 360
        Me.tChart1.Aspect.RotationFloat = 360
        Me.tChart1.Aspect.ZOffset = 0
        '
        '
        '
        '
        '
        '
        Me.tChart1.Axes.Bottom.MaximumOffset = 30
        Me.tChart1.Axes.Bottom.MinimumOffset = 30
        '
        '
        '
        Me.tChart1.Axes.Bottom.Title.Transparent = True
        '
        '
        '
        '
        '
        '
        Me.tChart1.Axes.Depth.Title.Transparent = True
        '
        '
        '
        '
        '
        '
        Me.tChart1.Axes.DepthTop.Title.Transparent = True
        '
        '
        '
        Me.tChart1.Axes.Left.MaximumOffset = 37
        '
        '
        '
        Me.tChart1.Axes.Left.Title.Transparent = True
        '
        '
        '
        '
        '
        '
        Me.tChart1.Axes.Right.Title.Transparent = True
        '
        '
        '
        '
        '
        '
        Me.tChart1.Axes.Top.Title.Transparent = True
        '
        '
        '
        Me.tChart1.Header.Lines = New String() {"tChart1"}
        Me.tChart1.Location = New System.Drawing.Point(0, 0)
        '
        '
        '
        '
        '
        '
        Me.tChart1.Panel.Brush.Color = System.Drawing.Color.FromArgb(CType(CType(254, Byte), Integer), CType(CType(255, Byte), Integer), CType(CType(255, Byte), Integer), CType(CType(255, Byte), Integer))
        Me.tChart1.Series.Add(Me.bar1)
        Me.tChart1.Size = New System.Drawing.Size(432, 212)
        Me.tChart1.TabIndex = 2
        '
        '
        '
        '
        '
        '
        Me.tChart1.Walls.Bottom.Size = 5
        '
        '
        '
        '
        '
        '
        Me.tChart1.Walls.Left.Brush.Color = System.Drawing.Color.FromArgb(CType(CType(254, Byte), Integer), CType(CType(232, Byte), Integer), CType(CType(232, Byte), Integer), CType(CType(255, Byte), Integer))
        Me.tChart1.Walls.Left.Size = 10
        '
        '
        '
        '
        '
        '
        Me.tChart1.Walls.Right.Brush.Color = System.Drawing.Color.FromArgb(CType(CType(254, Byte), Integer), CType(CType(192, Byte), Integer), CType(CType(192, Byte), Integer), CType(CType(192, Byte), Integer))
        Me.tChart1.Walls.Right.Size = 10
        Me.tChart1.Walls.Right.Visible = True
        '
        'chartContainer
        '
        Me.chartContainer.Location = New System.Drawing.Point(0, 105)
        Me.chartContainer.Size = New System.Drawing.Size(432, 212)
        '
        'hScrollBar2
        '
        Me.hScrollBar2.LargeChange = 1
        Me.hScrollBar2.Location = New System.Drawing.Point(131, 10)
        Me.hScrollBar2.Maximum = 180
        Me.hScrollBar2.Name = "hScrollBar2"
        Me.hScrollBar2.Size = New System.Drawing.Size(134, 14)
        Me.hScrollBar2.TabIndex = 3
        Me.hScrollBar2.Value = 90
        '
        'label1
        '
        Me.label1.AutoSize = True
        Me.label1.Location = New System.Drawing.Point(52, 10)
        Me.label1.Name = "label1"
        Me.label1.Size = New System.Drawing.Size(79, 13)
        Me.label1.TabIndex = 0
        Me.label1.Text = "&Rotation angle:"
        Me.label1.TextAlign = System.Drawing.ContentAlignment.TopRight
        '
        'label2
        '
        Me.label2.AutoSize = True
        Me.label2.Location = New System.Drawing.Point(272, 10)
        Me.label2.Name = "label2"
        Me.label2.Size = New System.Drawing.Size(25, 13)
        Me.label2.TabIndex = 2
        Me.label2.Text = "360"
        Me.label2.TextAlign = System.Drawing.ContentAlignment.TopRight
        '
        'bar1
        '
        '
        '
        '
        Me.bar1.Brush.Color = System.Drawing.Color.Red
        '
        '
        '
        Me.bar1.Brush.Gradient.EndColor = System.Drawing.Color.Red
        Me.bar1.Brush.Gradient.StartColor = System.Drawing.Color.FromArgb(CType(CType(254, Byte), Integer), CType(CType(255, Byte), Integer), CType(CType(255, Byte), Integer), CType(CType(255, Byte), Integer))
        Me.bar1.Brush.Gradient.Visible = True
        Me.bar1.Color = System.Drawing.Color.Red
        Me.bar1.ColorEach = False
        '
        '
        '
        '
        '
        '
        Me.bar1.Marks.Callout.ArrowHead = Steema.TeeChart.Styles.ArrowHeadStyles.None
        Me.bar1.Marks.Callout.ArrowHeadSize = 8
        '
        '
        '
        Me.bar1.Marks.Callout.Brush.Color = System.Drawing.Color.Black
        Me.bar1.Marks.Callout.Distance = 0
        Me.bar1.Marks.Callout.Draw3D = False
        Me.bar1.Marks.Callout.Length = 20
        Me.bar1.Marks.Callout.Series = Me.bar1
        Me.bar1.Marks.Callout.Style = Steema.TeeChart.Styles.PointerStyles.Rectangle
        Me.bar1.Marks.Callout.Visible = False
        Me.bar1.Marks.Series = Me.bar1
        Me.bar1.Title = "bar1"
        '
        '
        '
        Me.bar1.XValues.DataMember = "X"
        Me.bar1.XValues.Order = Steema.TeeChart.Styles.ValueListOrder.Ascending
        '
        '
        '
        Me.bar1.YValues.DataMember = "Bar"
        '
        'Canvas_180Rotation
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(432, 317)
        Me.Name = "Canvas_180Rotation"
        Me.panel1.ResumeLayout(False)
        Me.panel1.PerformLayout()
        Me.chartContainer.ResumeLayout(False)
        Me.ResumeLayout(False)
        Me.PerformLayout()

    End Sub 'InitializeComponent 
#End Region


    Private Sub hScrollBar2_Scroll(ByVal sender As Object, ByVal e As System.Windows.Forms.ScrollEventArgs) Handles hScrollBar2.Scroll
        If hScrollBar2.Value < 90 Then
            tChart1.Aspect.Rotation = 270 + hScrollBar2.Value
        Else
            tChart1.Aspect.Rotation = hScrollBar2.Value - 90
        End If
        label2.Text = tChart1.Aspect.Rotation.ToString()
    End Sub 'hScrollBar2_Scroll
End Class 'Canvas_180Rotation

File: WinForms/NET 6/NetDemoNET60/Canvas/Canvas_180Rotation.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;

using System.Windows.Forms;


namespace Steema.TeeChart.Samples
{
    public class Canvas_180Rotation : Samples.BaseForm
    {
        private HScrollBar hScrollBar2;
        private Label label1;
        private Label label2;
        private Steema.TeeChart.Styles.Bar bar1;
        private IContainer components = null;

        public Canvas_180Rotation()
        {
            // This call is required by the Windows Form Designer.
            InitializeComponent();
            tChart1.Aspect.View3D = true;
            tChart1.Aspect.Orthogonal = false;


            bar1.FillSampleValues(4);
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(disposing);
        }

        #region Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            Drawing.Cursor cursor1 = new Drawing.Cursor();
            ComponentResourceManager resources = new ComponentResourceManager(typeof(Canvas_180Rotation));
            hScrollBar2 = new HScrollBar();
            label1 = new Label();
            label2 = new Label();
            bar1 = new Styles.Bar();
            panel1.SuspendLayout();
            chartContainer.SuspendLayout();
            SuspendLayout();
            // 
            // textBox1
            // 
            textBox1.Size = new Size(432, 89);
            textBox1.TabIndex = 0;
            textBox1.Text = "Chart can be fully rotated along the horizontal (x) axis.\r\nIn previous versions it allowed 180 degree rotation only.";
            // 
            // panel1
            // 
            panel1.Controls.Add(label2);
            panel1.Controls.Add(label1);
            panel1.Controls.Add(hScrollBar2);
            panel1.Location = new Point(0, 89);
            panel1.Size = new Size(432, 40);
            panel1.TabIndex = 1;
            // 
            // tChart1
            // 
            // 
            // 
            // 
            // 
            // 
            // 
            tChart1.Header.Font.Size = 14;
            tChart1.Header.Font.SizeFloat = 14F;
            tChart1.Header.Lines = new string[] { "tChart1" };
            // 
            // 
            // 
            tChart1.Legend.CheckBoxes = false;
            // 
            // 
            // 
            // 
            // 
            // 
            tChart1.Panel.Brush.Color = Color.FromArgb(254, 255, 255, 255);
            // 
            // 
            // 
            tChart1.Panel.Brush.Gradient.EndColor = Color.FromArgb(254, 21, 60, 89);
            tChart1.Panel.Brush.Gradient.MiddleColor = Color.FromArgb(254, 255, 255, 255);
            tChart1.Panel.Brush.Gradient.Sigma = true;
            tChart1.Panel.Brush.Gradient.SigmaFocus = 1F;
            tChart1.Panel.Brush.Gradient.SigmaScale = 0.8F;
            tChart1.Panel.Brush.Gradient.StartColor = Color.FromArgb(254, 255, 255, 255);
            tChart1.Panel.Brush.Gradient.Visible = true;
            tChart1.Series.Add(bar1);
            tChart1.Size = new Size(432, 188);
            tChart1.TabIndex = 2;
            // 
            // 
            // 
            // 
            // 
            // 
            tChart1.Walls.Bottom.Size = 5;
            // 
            // 
            // 
            // 
            // 
            // 
            tChart1.Walls.Left.Brush.Color = Color.FromArgb(254, 232, 232, 255);
            tChart1.Walls.Left.Size = 10;
            // 
            // 
            // 
            // 
            // 
            // 
            tChart1.Walls.Right.Brush.Color = Color.FromArgb(254, 192, 192, 192);
            tChart1.Walls.Right.Size = 10;
            tChart1.Walls.Right.Visible = true;
            // 
            // chartContainer
            // 
            chartContainer.Location = new Point(0, 129);
            chartContainer.Size = new Size(432, 188);
            // 
            // hScrollBar2
            // 
            hScrollBar2.LargeChange = 1;
            hScrollBar2.Location = new Point(157, 12);
            hScrollBar2.Maximum = 360;
            hScrollBar2.Name = "hScrollBar2";
            hScrollBar2.Size = new Size(161, 18);
            hScrollBar2.TabIndex = 3;
            hScrollBar2.Scroll += hScrollBar2_Scroll;
            // 
            // label1
            // 
            label1.AutoSize = true;
            label1.Location = new Point(62, 12);
            label1.Name = "label1";
            label1.Size = new Size(87, 15);
            label1.TabIndex = 0;
            label1.Text = "&Rotation angle:";
            label1.TextAlign = ContentAlignment.TopRight;
            // 
            // label2
            // 
            label2.AutoSize = true;
            label2.Location = new Point(326, 12);
            label2.Name = "label2";
            label2.Size = new Size(25, 15);
            label2.TabIndex = 2;
            label2.Text = "360";
            label2.TextAlign = ContentAlignment.TopRight;
            // 
            // bar1
            // 
            bar1.BarRound = Styles.BarRounding.AtValue;
            // 
            // 
            // 
            // 
            // 
            // 
            bar1.Brush.Gradient.Visible = true;
            bar1.Color = Color.Red;
            bar1.Cursor = cursor1;
            // 
            // 
            // 
            bar1.Marks.ArrowLength = 20;
            bar1.Title = "bar1";
            // 
            // 
            // 
            bar1.XValues.DataMember = "X";
            bar1.XValues.Order = Styles.ValueListOrder.Ascending;
            bar1.XValues.Value = new double[] { 0D, 1D, 2D, 3D, 4D, 5D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D };
            // 
            // 
            // 
            bar1.YValues.DataMember = "Bar";
            bar1.YValues.Value = new double[] { 51D, 169D, 257D, 199D, 281D, 253D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D, 0D };
            // 
            // Canvas_180Rotation
            // 
            AutoScaleBaseSize = new Size(6, 16);
            ClientSize = new Size(432, 317);
            Name = "Canvas_180Rotation";
            panel1.ResumeLayout(false);
            panel1.PerformLayout();
            chartContainer.ResumeLayout(false);
            ResumeLayout(false);
            PerformLayout();
        }
        #endregion

        private void hScrollBar2_Scroll(object sender, ScrollEventArgs e)
        {
            tChart1.Aspect.Rotation = hScrollBar2.Value;
            label2.Text = tChart1.Aspect.Rotation.ToString();
        }
    }
}

